test/prob/discrete_range/discrete_range_ccdf_log_test.hpp

```diff
@@ -0,0 +1,65 @@
+// Arguments: Ints, Ints, Ints
+#include <stan/math/prim.hpp>
+
+using stan::math::var;
+using std::vector;
+
+class AgradCcdfLogDiscreteRange : public AgradCcdfLogTest {
+ public:
+  void valid_values(vector<vector<double>>& parameters,
+                    vector<double>& ccdf_log) {
+    vector<double> param(3);
+
+    param[0] = 3;  // y
+    param[1] = 1;  // lower
+    param[2] = 5;  // upper
+    parameters.push_back(param);
+    ccdf_log.push_back(-0.916290731874154995573);  // expected ccdf_log
+
+    param[0] = 9;   // y
+    param[1] = 5;   // lower
+    param[2] = 15;  // upper
+    parameters.push_back(param);
+    ccdf_log.push_back(-0.606135803570315601085);  // expected ccdf_log
+
+    param[0] = 0;   // y
+    param[1] = -4;  // lower
+    param[2] = 5;   // upper
+    parameters.push_back(param);
+    ccdf_log.push_back(-0.693147180559945286227);  // expected ccdf_log
+  }
+
+  void invalid_values(vector<size_t>& /*index*/, vector<double>& /*value*/) {
+    // y
+
+    // lower
+
+    // upper
+  }
+
+  bool has_lower_bound() { return false; }
```

Comment:
I'm not exactly sure how these tests work, but this distribution has a lower and upper bound.

Reply:
Those are used for distributions that have a fixed support, say the `beta` that is in [0,1]. There is a kludge for `uniform` that sets only the upper bound. So I'm not sure it's worth trying to use those here, I can add some prim tests about boundary behaviour.